base/dict.jl

```diff
@@ -189,6 +189,12 @@ function grow_to!{K,V}(dest::Associative{K,V}, itr, st)
     return dest
 end
 
+
+"""
+    similar(d::Dict)
+
+Create an empty dictionary with the same key and value types as `d`.
+"""
 similar{K,V}(d::Dict{K,V}) = Dict{K,V}()
 similar{K,V}(d::Dict, ::Type{Pair{K,V}}) = Dict{K,V}()
```

Comment:
I am not sure I understand the meaning of the second expression: isn't all information to create the dict already contained in `d`?